nucliadb/src/nucliadb/common/cache.py

```diff
@@ -139,46 +126,42 @@ def __init__(self, cache_size: int = 128):
 # Cache management
 
 
-# Get or create a resource cache specific to the current asyncio task (and all
-# its subtasks). If you spawn subtasks that use this cache, make sure to create
-# it in the parent task, otherwise each subtask will have its own independent
-# cache instance
-def get_or_create_resource_cache(clear: bool = False) -> ResourceCache:
-    cache: Optional[ResourceCache] = rcache.get()
-    if cache is None or clear:
-        cache = ResourceCache()
-        rcache.set(cache)
-    return cache
-
-
 def get_resource_cache() -> Optional[ResourceCache]:
     return rcache.get()
 
 
-def set_resource_cache() -> None:
-    cache = ResourceCache()
-    rcache.set(cache)
+def get_extracted_text_cache() -> Optional[ExtractedTextCache]:
+    return etcache.get()
 
 
-# Delete resource cache and all its content
-def delete_resource_cache() -> None:
-    cache = rcache.get()
-    if cache is not None:
-        rcache.set(None)
-        cache.clear()
+@contextlib.contextmanager
+def _use_cache(klass: type[Cache], context_var: ContextVar, /, **kwargs):
+    """Context manager that manages a context var cache. It's responsible of
+    cache creation and cleanup.
 
+    Note the configured cache is specific to the current asyncio task (and all
+    its subtasks). If you spawn subtasks that should share a cache, make sure
+    the parent task is the one using this decorator, otherwise, each subtask
+    will use its own independent cache instance
 
-def get_extracted_text_cache() -> Optional[ExtractedTextCache]:
-    return etcache.get()
+    Do not use the cache object outside the scope of this context manager!
+    Otherwise, metrics and cleanup could get wrong.
+
+    """
+    cache = klass(**kwargs)
+    token = context_var.set(cache)
+    yield cache
+    context_var.reset(token)
+    cache.clear()
```

Comment:
can you add a try:finally and make sure the cache is always cleared? also when there is an error